lib/charms/opensearch/v0/opensearch_tls.py

```diff
@@ -210,23 +210,30 @@ def _on_certificate_available(self, event: CertificateAvailableEvent) -> None:
         if not self.charm.unit.is_leader() and scope == Scope.APP:
             return
 
-        if not self.ca_rotation_complete_in_cluster():
```

Comment:
This is the main change. This check here filters out any `cert-available` events if at least one single unit in the entire cluster started the rotation.

That means we have a race. If we start a rotation, some units will cross this line and eventually get into the rotation part of this function. Once they get there and finish their own flow, then this check will start to return `false` for all the other units that did not yet get a `cert-available` event.

I do not see the value of this code, so I just removed it.

Reply:
So, that is interesting. It means the onl real place we have where `store_new_tls_resources`  **really** gets applied then [is here](https://github.com/canonical/opensearch-operator/blob/405725f7d87acc2f37e3f91027116b18c54d1153/lib/charms/opensearch/v0/opensearch_tls.py#L645-L648), would that be correct?

Reply:
Yes, this is where the certs are actually applied.